### PR TITLE
Feature/72 remove unreachable code

### LINKED ISSUE #72: Remove unreachable code in KafkaStreamReader
Currently, the else-block in the following snippet in `KafkaStreamReader` is unreachable. 
```
    val optionalKeys = getKeysFromPrefix(configuration, rootFactoryOptionalConfKey)

    val extraConfs = optionalKeys.foldLeft(Map[String,String]()) {
      case (map,securityKey) =>
        getOrNone(securityKey, configuration) match {
          case Some(value) => map + (tweakOptionKeyName(securityKey) -> value)
          case None => map
        }
    }

    if (extraConfs.isEmpty || extraConfs.size == optionalKeys.size) {
      extraConfs
    }
    else {
      logger.warn(s"Assuming no security settings, since some appear to be missing: {${findMissingKeys(optionalKeys, extraConfs)}}")
      Map[String,String]()
    }
```
`optionalKeys` gets keys from the `configuration` and `extraConfs` gets the `optionalKeys` again from `configuration`. `configuration` does not change between the two calls. Therefore, `extraConfs` will always have the same size as `optionalKeys`

Moreover, even if some settings were missing, they should not be removed, as this could lead to unexpected error messages downstream.
Therefore, the if-else-block should be removed.

### PR DESCRIPTION
**Pull request by kevinwallimann**
Closes #72

### CODE REVIEW

**Comment by Zejnilovic** [comment]
Just beware. This branch seems to be cut off the #73 PR branch

**Comment by kevinwallimann** [comment]
> Just beware. This branch seems to be cut off the #73 PR branch

Yes, that's intended. Otherwise, I'd have to wait until the other PR is merged

File: ingestor-default/src/test/scala/za/co/absa/hyperdrive/ingestor/implementation/writer/parquet/TestParquetStreamWriter.scala [hunk 1]
```diff
@@ -57,47 +57,47 @@ class TestParquetStreamWriter extends FlatSpec with MockitoSugar {
...
+    invokeWriter(dataStreamWriter, offsetManager, Map())
     verify(offsetManager).configureOffsets(dataStreamWriter, configuration)
   }
 
   it should "start DataStreamWriter" in {
     val dataStreamWriter = getDataStreamWriter
     val offsetManager = getOffsetManager(dataStreamWriter)
 
-    invokeWriter(dataStreamWriter, offsetManager, None)
+    invokeWriter(dataStreamWriter, offsetManager, Map())
```
[the remark this answers is not in the document]
Comment (kevinwallimann) [reply ingestor-default/src/test/scala/za/co/absa/hyperdrive/ingestor/implementation/writer/parquet/TestParquetStreamWriter.scala]: I'll keep it like this. Less verbose

File: ingestor-default/src/test/scala/za/co/absa/hyperdrive/ingestor/implementation/writer/parquet/TestParquetStreamWriter.scala [hunk 2]
```diff
@@ -57,47 +57,47 @@ class TestParquetStreamWriter extends FlatSpec with MockitoSugar {
     val dataFrame = mock[DataFrame]
     when(dataFrame.writeStream).thenReturn(dataStreamWriter)
 
-    val writer = new ParquetStreamWriter(parquetDestination.getAbsolutePath, None)
+    val writer = new ParquetStreamWriter(parquetDestination.getAbsolutePath, Map())
     assertThrows[IllegalArgumentException](writer.write(dataFrame, offsetManager = null))
   }
 
-  it should "set format as 'parquet'" in {
+  "aldsfjk" should "set format as 'parquet'" in {
```
[the remark this answers is not in the document]
Comment (kevinwallimann) [reply ingestor-default/src/test/scala/za/co/absa/hyperdrive/ingestor/implementation/writer/parquet/TestParquetStreamWriter.scala]: oh no. *facepalm*

**Pull request merged**